pandas/core/frame.py

```diff
@@ -7375,7 +7375,7 @@ def swaplevel(self, i: Axis = -2, j: Axis = -1, axis: Axis = 0) -> DataFrame:
             result.columns = result.columns.swaplevel(i, j)
         return result
 
-    def reorder_levels(self, order: Sequence[int | str], axis: Axis = 0) -> DataFrame:
+    def reorder_levels(self, order: Sequence[Hashable], axis: Axis = 0) -> DataFrame:
```

Comment:
I undid this change because I fear I would have to dig much deeper. Further functions called by reorder_levels are typed with `int | str` where actually a lot of other types work as well - but I am not sure whether this is accidentally supported or not. It seems you can have a float label, for example - not that I've come across a use case.

In order not to dig deeper here, I got rid of this change and am limiting this commit to value_counts, stack and unstack where the changes should hopefully be uncontroversial. (lets hope I did the ignore comment correctly)